packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -0,0 +1,444 @@
+                  onPointerMove: _handlePointerMove,
+                  onPointerUp: _handlePointerUp,
+                  onPointerCancel: _handlePointerCancel,
+                  behavior: HitTestBehavior.opaque,
+                  child: isSelectMode
+                      ? new IgnorePointer(
+                          key: _ignorePointerKey,
+                          ignoringSemantics: false,
+                          child: widget.child)
+                      : widget.child)),
+          new Positioned(
+            left: _kInspectButtonMargin,
+            bottom: _kInspectButtonMargin,
+            child: isSelectMode
+                ? new Container()
+                : new FloatingActionButton(
+                    child: const Icon(Icons.search),
+                    onPressed: _handleEnableSelect,
+                    mini: true,
+                  ),
+          ),
+        ]));
+  }
+}
+
+const double _kScreenEdgeMargin = 10.0;
+const double _kTooltipPadding = 5.0;
+const double _kInspectButtonMargin = 10.0;
+
+/// Interpret pointer up events within with this margin as indicating the
+/// pointer is moving off the device.
+const double _kOffScreenMargin = 1.0;
+
+// TODO(jacobr): merge with similar code in tooltip.dart.
+class _TooltipPositionCalculator {
+  _TooltipPositionCalculator({
+    this.target,
+    this.verticalOffset,
+    this.preferBelow,
+    this.size,
+    this.childSize,
+  }) {
+    // VERTICAL DIRECTION
+    final bool fitsBelow = target.dy + verticalOffset + childSize.height <=
+        size.height - _kScreenEdgeMargin;
+    final bool fitsAbove =
+        target.dy - verticalOffset - childSize.height >= _kScreenEdgeMargin;
+    _tooltipBelow =
+        preferBelow ? fitsBelow || !fitsAbove : !(fitsAbove || !fitsBelow);
+    double y;
+    if (_tooltipBelow)
+      y = math.min(
+          target.dy + verticalOffset, size.height - _kScreenEdgeMargin);
+    else
+      y = math.max(
+          target.dy - verticalOffset - childSize.height, _kScreenEdgeMargin);
+    // HORIZONTAL DIRECTION
+    final double normalizedTargetX =
+        target.dx.clamp(_kScreenEdgeMargin, size.width - _kScreenEdgeMargin);
+    double x;
+    if (normalizedTargetX > size.width - _kScreenEdgeMargin - childSize.width) {
+      // Make room for content to the right.
+      x = size.width - _kScreenEdgeMargin - childSize.width;
+    } else {
+      x = normalizedTargetX;
+    }
+    _childPosition = new Offset(x, y);
+  }
+
+  final Offset target;
+  final double verticalOffset;
+  final bool preferBelow;
+  final Size size;
+  final Size childSize;
+
+  Offset _childPosition;
+  bool _tooltipBelow;
+
+  bool get tooltipBelow => _tooltipBelow;
+  Offset get childPosition => _childPosition;
+}
+
+// FOR DISCUSSION: this is clearly not quite right: should I even even bother
+// listening for changes? This code is copied from semantics_debugger where
+// it made sense.
+class _WidgetClient extends ChangeNotifier {
```

Comment:
I don't. This code is pretty bogus. It only made sense for the semantic debugger. Is thee way I can get similar change notifications for Widget changes?